spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkConfParser.java

```diff
@@ -70,6 +70,14 @@ public DurationConfParser durationConf() {
     return new DurationConfParser();
   }
 
+  private static CaseInsensitiveStringMap asCaseInsensitiveStringMap(Map<String, String> map) {
```

Comment:
Prior to split of Spark 2 and Spark 3, we had to use `Map<String, String>` as Spark 3 used one class for options and Spark 2 used another one. It is always `CaseInsensitiveStringMap` in Spark 3 but I am checking here to be safe.